src/scheduler/workers/unpublish/delete/index.ts

```diff
@@ -0,0 +1,94 @@
+import { startOfToday, subYears } from "date-fns";
+import { prisma } from "scheduler/prismaClient";
+import { schedulerLogger } from "scheduler/logger";
+import type { ListItem, Event } from "shared/types";
+
+export default async function deleteItemsAfterAYear() {
+  const logger = schedulerLogger.child({ method: "deleteItemsAfterAYear" });
+
+  try {
+    const dateOneYearAgo = subYears(startOfToday(), 1);
+    const itemsUnpublishedByAR = await prisma.listItem.findMany({
+      where: {
+        status: "ANNUAL_REVIEW_OVERDUE",
+        isAnnualReview: false,
+        isPublished: false,
+        AND: [
+          {
+            history: {
+              some: {
+                type: "UNPUBLISHED",
+                time: {
+                  lte: dateOneYearAgo,
+                },
+              },
+            },
+          },
+          {
+            history: {
+              some: {
+                type: "ANNUAL_REVIEW_OVERDUE",
+                time: {
+                  lte: dateOneYearAgo,
+                },
+              },
+            },
+          },
+          {
+            history: {
+              none: {
+                type: "PUBLISHED",
+                time: {
+                  gte: dateOneYearAgo,
+                },
+              },
+            },
+          },
+        ],
+      },
+      include: {
+        history: {
+          orderBy: {
+            time: "desc",
+          },
+        },
+      },
```

Comment:
```suggestion
      include: {
        history: {
          take: 1,
          where: {
            type: "UNPUBLISHED",
            time: {
              lte: dateOneYearAgo,
            },
          },
          orderBy: {
            time: "desc",
          },
        },
      },
```
`take: 1` - returns only the first item in history ordered by desc, and is "UNPUBLISHED" and time lte dateOneYearAgo.

That way you can remove your other iterator, `getUnpublishedIndex`.

all you have to do is item.history[0].

if you want to shorten the lines this query runs over, you can store some of these clauses in vars.
```
const unpublishedAYearAgo: `EventWhereInput` = {
  type: "UNPUBLISHED", 
  time: ..
}
```

Reply:
Ah nice, I didn't know that 👍🏿